### PR TITLE
TST: Add int32 input testcase for cholesky.

### PR DESCRIPTION
**Pull request by Qiyu8**
`np.linalg.cholesky` accept `int32` input , But the testcase only provide float and complex input.

### CODE REVIEW

**Comment by eric-wieser** [comment]
This doesn't strike me as a particularly useful tests, since numpy just casts all inputs to these functions to `double` anyway.

Perhaps more useful would be a unit test for the internal `np.linalg._commonType` helper, which dictates the rules for those conversions, which are applied to all linalg functions.

**Comment by Qiyu8** [comment]
I found that there was an interesting debate on how to test a widely used private function in [#12201](https://github.com/numpy/numpy/pull/12201), One view is to directly test the private function, another view is to test through public function, IMO, private methods should be tested through public interfaces, If the private method is complex, extract as an util function to test. The reason are as follows:

- Although test the private method provides a sense of security and robustness, But It destroy the encapsulation, expose any privacy is not good in most cases.
- The only purpose of the test is to ensure that the public interface works normally, So pursuite for 100% coverage is not the goal. 
- There are better ways to remove dead code, writing tons test case of private function cost a lot. 

for this `np.linalg._commonType` case, this function is not complex, I will just add more public functions to trigger this internal function. Any other thoughts?

**Comment by eric-wieser** [comment]
Good find on that issue, the discussion there is very relevant. I think my opinion has changed a little since I wrote my comments there.

> If the private method is complex, extract as an util function to test.

The flaw in this argument is that anything that becomes a public util function has to be supported for years, since it becomes fair game for downstream packages to use it.

> Although test the private method provides a sense of security and robustness, But It destroy the encapsulation, expose any privacy is not good in most cases.

I think as long as the test is clearly marked as a test of a private function, it does not count as exposing private details.

**Comment by Qiyu8** [comment]
considering the CI runtime cost, Let's just test the internal function. 😄

File: numpy/linalg/tests/test_linalg.py [hunk 1]
```diff
@@ -2043,10 +2043,9 @@ def test_tensorinv_result(self):
 
 def test_unsupported_commontype():
     # linalg gracefully handles unsupported type
-    arr = np.array([[1, -2], [2, 5]], dtype='float16')
+    arr = np.array(['3.5', '2.'], dtype='float16')
```
Comment (rossbar) [inline numpy/linalg/tests/test_linalg.py:2046]: What is the motivation behind the change in the array values? Either one gives an array with a float16 dtype and the string values seem strange to me.

**Review by rossbar — COMMENTED** [review]
I'm having trouble connecting the discussion to the changes. It looks like the only pertinent change is that `test_unsupported_commontype` now calls the private `_commonType` function directly rather than calling it indirectly via `cholesky`. I have trouble understanding the motivation for the other changes though. Is this maybe a WIP?

File: numpy/linalg/tests/test_linalg.py (hunk 1 above)
Comment (Qiyu8) [reply numpy/linalg/tests/test_linalg.py:2046]: @rossbar this is indeed a trivial change, which leads to a trivial but interesting argument, anyway I will close this for PR mangement.
